tests/ui/components/filters/filterboxcomponent.js

```diff
@@ -148,31 +149,48 @@ describe('filter box component', () => {
 
     child1._updateOption(0, true);
-    expect(child1.getFilterNode()).toEqual(nodes1[0]);
-    expect(setStaticFilterNodes.mock.calls[4][1]).toEqual(child0.getFilterNode());
-    expect(setStaticFilterNodes.mock.calls[5][1]).toEqual(child1.getFilterNode());
+    expect(child1.getFilterNode().getFilter()).toEqual(nodes1[0].getFilter());
+    expect(child1.getFilterNode().getMetadata()).toEqual(nodes1[0].getMetadata());
+    expect(setStaticFilterNodes.mock.calls[4][1].getFilter()).toEqual(child0.getFilterNode().getFilter());
+    expect(setStaticFilterNodes.mock.calls[4][1].getMetadata()).toEqual(child0.getFilterNode().getMetadata());
+    expect(setStaticFilterNodes.mock.calls[5][1].getFilter()).toEqual(child1.getFilterNode().getFilter());
+    expect(setStaticFilterNodes.mock.calls[5][1].getMetadata()).toEqual(child1.getFilterNode().getMetadata());
 
     child1._updateOption(3, true);
-    expect(child1.getFilterNode()).toEqual(FilterNodeFactory.and(nodes1[0], nodes1[3]));
-    expect(setStaticFilterNodes.mock.calls[6][1]).toEqual(child0.getFilterNode());
-    expect(setStaticFilterNodes.mock.calls[7][1]).toEqual(child1.getFilterNode());
+    expect(child1.getFilterNode().getFilter()).toEqual(FilterNodeFactory.and(nodes1[0], nodes1[3]).getFilter());
+    expect(child1.getFilterNode().getMetadata()).toEqual(FilterNodeFactory.and(nodes1[0], nodes1[3]).getMetadata());
+    expect(setStaticFilterNodes.mock.calls[6][1].getFilter()).toEqual(child0.getFilterNode().getFilter());
+    expect(setStaticFilterNodes.mock.calls[6][1].getMetadata()).toEqual(child0.getFilterNode().getMetadata());
+    expect(setStaticFilterNodes.mock.calls[7][1].getFilter()).toEqual(child1.getFilterNode().getFilter());
+    expect(setStaticFilterNodes.mock.calls[7][1].getMetadata()).toEqual(child1.getFilterNode().getMetadata());
 
     child1._updateOption(4, true);
-    expect(child1.getFilterNode()).toEqual(
-      FilterNodeFactory.and(nodes1[0], FilterNodeFactory.or(nodes1[3], nodes1[4]))
+    expect(child1.getFilterNode().getFilter()).toEqual(
+      FilterNodeFactory.and(nodes1[0], FilterNodeFactory.or(nodes1[3], nodes1[4])).getFilter()
     );
-    expect(setStaticFilterNodes.mock.calls[8][1]).toEqual(child0.getFilterNode());
-    expect(setStaticFilterNodes.mock.calls[9][1]).toEqual(child1.getFilterNode());
+    expect(child1.getFilterNode().getMetadata()).toEqual(
+      FilterNodeFactory.and(nodes1[0], FilterNodeFactory.or(nodes1[3], nodes1[4])).getMetadata()
+    );
+    expect(setStaticFilterNodes.mock.calls[8][1].getFilter()).toEqual(child0.getFilterNode().getFilter());
+    expect(setStaticFilterNodes.mock.calls[8][1].getMetadata()).toEqual(child0.getFilterNode().getMetadata());
+    expect(setStaticFilterNodes.mock.calls[9][1].getFilter()).toEqual(child1.getFilterNode().getFilter());
+    expect(setStaticFilterNodes.mock.calls[9][1].getMetadata()).toEqual(child1.getFilterNode().getMetadata());
```

Comment:
This test is really long, with a ton of assertions. That seems like a bit of a code smell to me. It suggests a couple possibilities:
- This one test is used to verify too many things and should be split apart.
- Some of these assertions are redundant or overlap. 

With our Jest tests, there's a bit of a balance. We want full test coverage, but we want to avoid any overlap between tests and assertions as much as possible. The more tests there are and the larger they are, the slower our unit test suite is to run. So, we want to be judicious about the new assertions we add or the new tests we add. We want to ensure that by adding something, we are expanding our coverage in a meaningful way.

Reply:
Yeah looking back when I have something like
`let actualNode = blah; let expectedNode = blah`
and then be continually changing what the expected and actual values are equal to, it can probably be easily separated into multiple tests. I think my tests are still a little too complicated but things should be much better now